### PR TITLE
Added logging

### PR DESCRIPTION
**Pull request by kafoso**
I added logging:

1. **Access** log: System-only*. Logs all requests.
2. **Error** log: System-only*. Logs all uncaught Errors and Exceptions, and "500 Internal Server Error" events.
3. **Manual** log: The log we use within the MVC framework to log e.g. debug messages.

*System-only means that only the framework should access and be allowed to write to these logs. Use the "manual" log for development purposes.

Next PR: https://github.com/Starbow/website/pull/10

**Please don't close or merge this pull request; let the person who created the pull request do this.**

### CODE REVIEW

File: server/bootstrap/logs.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        new winston.transports.File(subConfig.file)
+      ],
+      exitOnError: false
+  });
+};
+
+module.exports.init = function(config){
+  module.exports.access = getMorganLog(config.log.access);
+  module.exports.error = getWinstonLog(config.log.error);
+  module.exports.manual = getWinstonLog(config.log.manual);
```
Comment (kafoso) [inline server/bootstrap/logs.js]: The reason we use two different log libraries:

1. `morgan` is very light-weight. Better for accessing logging, which is done on every request.
2. `winston` is more complex and robust, and with better stack trace handling, making it better for logging severe errors.

Comment (Sikian) [inline server/bootstrap/logs.js]: Thanks for the explanation :)

File: server/mvc/log.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports.init = function(log){
```
Comment (Sikian) [inline server/mvc/log.js]: I guess this will have more stuff inside, as I don't really see the point here :octopus:

Comment (kafoso) [inline server/mvc/log.js]: No. This is just an abstraction layer on top of a pre-configured `winston` module. It works out-of-the-box.

All logs (incl. this "manual" log, as the `var log` refers to) are configured in [`server/bootstrap/logs.js`](https://github.com/Starbow/website/blob/4df559086c2b5fef4a87e2b2e02199c450f60050/server/bootstrap/logs.js).

Preferably, we should never reference anything placed in `server/bootstrap` when working with logic contained within the `server/mvc` folders, i.e. controllers and models. Bootstrap is for configuring the server, exclusively.

All models receive the full `config` Object upon initialization (see [here](https://github.com/Starbow/website/blob/4df559086c2b5fef4a87e2b2e02199c450f60050/server/mvc/models.js#L5)). However, the `config` Object does not contain logic; only data.

This means the fully configured logs `access` and `error` are completely off-limits in the MVC logic, quite intentionally.

I've made a booboo and done incorrectly in [`server/mvc/controllers/ProfileController.js#L1`](https://github.com/Starbow/website/blob/4df559086c2b5fef4a87e2b2e02199c450f60050/server/mvc/controllers/ProfileController.js#L1); the controller should have no awareness or access to configuration data, except through models. I'm going to fix this and move the logic to a model.

**TL;DR**
It's intentional. The `winston` log retrieved here is fully configured and will write log info to `manual.log`.

Comment (kafoso) [inline server/mvc/log.js]: I'm working on a ([DevExamples](https://github.com/Starbow/website/blob/4d6890a263e94767860ce52b9c78ab41018e619d/server/mvc/models/DevExamples.js)) on a different branch. I'll add some logic explaining this behavior.

Comment (Sikian) [inline server/mvc/log.js]: Thanks for the explanation, it now makes sense!

**Comment by Sikian** [comment]
I think we can go ahead and merge this. Solid logs! :D
